### PR TITLE
[HttpClient] Allow custom working directory in TestHttpServer

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

eases mocking by a lot :)

### CODE REVIEW

**Comment by ro0NL** [comment]
with this enabled we could think of new frameworkbundle config, to route all clients to the test server, so we guarantee no external traffic occurs

File: src/Symfony/Contracts/HttpClient/Test/TestHttpServer.php [hunk 1]
```diff
@@ -18,8 +18,13 @@ class TestHttpServer
 {
     private static $process = [];
 
-    public static function start(int $port = 8057): Process
+    /**
+     * @param string|null $workingDirectory
+     */
+    public static function start(int $port = 8057/**, $workingDirectory = null*/): Process
```
Comment (nicolas-grekas) [inline src/Symfony/Contracts/HttpClient/Test/TestHttpServer.php]:
```suggestion
    public static function start(int $port = 8057/*, string $workingDirectory = null*/): Process
```

**Review by nicolas-grekas — COMMENTED** [review]
needs something in the changelog

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Contracts/HttpClient/Test/TestHttpServer.php [hunk 2]
```diff
@@ -18,8 +18,13 @@ class TestHttpServer
...
     private static $process = [];
 
-    public static function start(int $port = 8057): Process
+    /**
+     * @param string|null $workingDirectory
+     */
+    public static function start(int $port = 8057/*, string $workingDirectory = null*/): Process
     {
+        $workingDirectory = \func_get_args()[1] ?? __DIR__.'/Fixtures/web';
+
```
Comment (nicolas-grekas) [inline src/Symfony/Contracts/HttpClient/Test/TestHttpServer.php]: from fabbot:

```suggestion
    public static function start(int $port = 8057/* , string $workingDirectory = null */): Process
    {
        $workingDirectory = \func_get_args()[1] ?? __DIR__.'/Fixtures/web';

```

**Review by javiereguiluz — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Pull request closed without merging**